clouddriver-google/src/main/groovy/com/netflix/spinnaker/clouddriver/google/deploy/handlers/BasicGoogleDeployHandler.groovy

```diff
@@ -534,4 +549,21 @@ class BasicGoogleDeployHandler implements DeployHandler<BasicGoogleDeployDescrip
       null
     }
   }
+
+  Map getUserData(BasicGoogleDeployDescription description, String serverGroupName,
+                  String instanceTemplateName, GoogleNamedAccountCredentials credentials) {
+    String customUserData = ''
+    if (description.userDataType && description.userData) {
+      if (description.userDataType.equals("URL")) {
+        task.updateStatus BASE_PHASE, "Resolving user data from url $description.userData..."
+        customUserData = description.userData.toURL()?.text
+      } else {
+        customUserData = description.userData
+      }
+    }
+    Map userData = googleUserDataProvider.getUserData(serverGroupName, instanceTemplateName,
```

Comment:
I guess that makes the semantics a little strange - type must be anything but "url" for any custom user data to be applied. It feels like that might be a bug in that PR.